blueprints/automation/tado_temp_offset.yaml

```diff
@@ -43,35 +45,30 @@ blueprint:
         boolean:
 
     night_switch:
-      name: 'Nighttime (optional)'
-      description: 'At nighttime the sync rate is reduced to every 2 hours'
-      default:
+      name: "Nighttime (optional)"
+      description: "At nighttime the sync rate is reduced to every 2 hours"
       selector:
         entity:
-          domain: 
+          domain:
             - binary_sensor
             - input_boolean
-          
+
   source_url: https://gist.github.com/clemensao/homeassistant/blob/main/blueprints/automation/tado_temp_offset.yaml
+
 variables:
-  target_tado: !input 'target_tado'
-  source_temp_sensor: !input 'source_temp_sensor'
-  heating_threshold: !input 'heating_threshold'
-  night_switch: !input 'night_switch'
-  trigger_sensor: !input 'trigger_sensor'
-  tado_temp: '{{ state_attr(target_tado, "current_temperature") | float(19) }}'
+  target_tado: !input "target_tado"
+  source_temp_sensor: !input "source_temp_sensor"
+  heating_threshold: !input "heating_threshold"
+  night_switch: !input "night_switch"
+  trigger_sensor: !input "trigger_sensor"
+  tado_temp: '{{ state_attr(target_tado, "current_temperature") | float }}'  # Fetch dynamically
   current_offset: '{{ state_attr(target_tado, "offset_celsius") }}'
-  actual_temp: '{{ states(source_temp_sensor) | float(19) }}'
-  run_flag: '{{ night_switch == none or (night_switch != none and is_state(night_switch, "off")) }}'
-  heating_rate: >
-    {% set entities = device_entities(device_id("climate.schlafzimmer") ) %}
-    {% for s in entities %}
-      {% if 'heating' in s %}
-        {{ states(s) }}
-      {% endif %}
-    {% endfor %}
+  actual_temp: "{{ states(source_temp_sensor) | float }}"
+  run_flag: '{{ night_switch is not none and is_state(night_switch, "off") }}'
```

Comment:
What will happen if night_switch is not set? Nighttime is an optional input! 
I admit "run_flag" is not a good variable name. It should be s.th. like day_flag

`run_flag: '{{ night_switch is none or (night_switch is not none and is_state(night_switch, "off")) }}' # flag for determining the daytime-mode`

Reply:
I'll have a play about and see what it the debug looks like